scylla/src/cloud/config.rs

```diff
@@ -74,21 +72,51 @@ impl CloudConfig {
 /// to connect to cloud nodes.
 #[derive(Debug)]
 pub(crate) struct AuthInfo {
-    key: PKey<Private>,
-    cert: X509,
+    tls: TlsInfo,
     #[allow(unused)]
     username: Option<String>,
     #[allow(unused)]
     password: Option<String>,
 }
 
-impl AuthInfo {
-    pub(crate) fn get_key(&self) -> &PKey<Private> {
-        &self.key
+#[derive(Debug)]
+pub(crate) enum TlsInfo {
+    #[cfg(feature = "openssl")]
+    OpenSsl {
+        key: openssl::pkey::PKey<openssl::pkey::Private>,
+        cert: openssl::x509::X509,
+    },
+    #[cfg(feature = "rustls")]
+    #[allow(dead_code)]
+    Rustls {
+        cert_chain: Vec<rustls::pki_types::CertificateDer<'static>>,
+        key: rustls::pki_types::PrivateKeyDer<'static>,
+    },
+}
+
+impl TlsInfo {
+    fn from_pem(cert: &[u8], key: &[u8]) -> Result<Self, TlsError> {
+        #[cfg(feature = "openssl")]
+        {
+            let cert = openssl::x509::X509::from_pem(cert)?;
+            let key = openssl::pkey::PKey::private_key_from_pem(key)?;
+            Ok(TlsInfo::OpenSsl { key, cert })
+        }
+
+        #[cfg(all(not(feature = "openssl"), feature = "rustls"))]
+        {
+            use rustls::pki_types::pem::PemObject;
+            let key = rustls::pki_types::PrivateKeyDer::from_pem_slice(key)?;
+            let cert_chain: Vec<_> = rustls::pki_types::CertificateDer::pem_slice_iter(cert)
+                .collect::<Result<_, _>>()?;
+            Ok(TlsInfo::Rustls { cert_chain, key })
+        }
     }
+}
```

Comment:
:thought_balloon:  This seems to work. In case of both `openssl` and `rustls` features being enabled, openssl is chosen. Fortunately, as this is the creation of `TlsInfo`, it's the only place where we have to do such logic based on features; in other places we're just matching on `TlsInfo`.